### PR TITLE
set process field on firebase steps field

### PR DESCRIPTION
**Pull request by kim-codefresh**

### CODE REVIEW

File: taskLogger/StepLogger.js [hunk 1]
```diff
@@ -262,6 +262,11 @@ class StepLogger extends EventEmitter {
         this.status = status;
         return this._reportStatus();
     }
+
+    setLogProcess(log) {
```
Comment (daniel-codefresh) [inline taskLogger/StepLogger.js:266]: hmm I'm not sure i like the name logProcess
I think that just "progress" or "stepProgress" would be more appropriate

File: taskLogger/composition/StepLogger.js [hunk 2]
```diff
@@ -111,6 +111,13 @@ class CompositionStepLogger extends BaseStepLogger {
 
     }
 
+    _reportLogProcess() {
+        this.loggers.forEach((logger) => {
```
Comment (daniel-codefresh) [inline taskLogger/composition/StepLogger.js:115]: not sure i recall, why are there many loggers? is it multiple loggers created for the same step? or are these loggers represent multiple steps?

File: taskLogger/test/StepLogger.unit.spec.js [hunk 3]
```diff
@@ -462,4 +463,15 @@ describe('Base StepLogger tests', () => {
 
     });
 
+    describe('setLogProcess', () => {
+
+        it('should set log process', () => {
+            const stepLogger = getStepLoggerInstance();
+            const logMsg = 'Validating connection to Docker daemon...';
+            stepLogger.setLogProcess(logMsg);
+            expect(stepLogger._reportLogProcess).to.have.been.calledWith();
```
Comment (daniel-codefresh) [inline taskLogger/test/StepLogger.unit.spec.js:472]: maybe lets also add a check on the progress field it self to verify that the state is changing

File: taskLogger/firebase/StepLogger.js [hunk 4]
```diff
@@ -107,6 +107,10 @@ class FirebaseStepLogger extends BaseStepLogger {
         this.stepRef.child('metrics').child('logs').child('total').set(this.logSize);
     }
 
+    _reportStepProgress() {
+        this.stepRef.child('process').set(this.stepProgress);
```
Comment (daniel-codefresh) [inline taskLogger/firebase/StepLogger.js:111]: lets also change the the name of the field in firebase

File: taskLogger/firebase/rest/StepLogger.js [hunk 5]
```diff
@@ -116,6 +116,13 @@ class FirebaseRestStepLogger extends FirebaseStepLogger {
             });
     }
 
+    _reportStepProgress() {
+        this.restClient.set(`${this.stepRef.ref().toString()}/process`, this.logProcess)
```
Comment (daniel-codefresh) [inline taskLogger/firebase/rest/StepLogger.js:120]: same thing here - lets rename the resource name in the path

Comment (daniel-codefresh) [reply taskLogger/firebase/rest/StepLogger.js:120]: also `this.logProcess` no longer exists - please make sure that everything still works after you finish with the renames, both in firebase and in mongo/redis(offlineLogging)

File: taskLogger/redis/StepLogger.js [hunk 6]
```diff
@@ -71,6 +71,10 @@ class RedisStepLogger extends BaseStepLogger {
         this.writter.child('metrics.logs.total').set(this.logSize);
     }
 
+    _reportStepProgress() {
+        this.writter.child('process').set(this.stepProgress);
```
Comment (daniel-codefresh) [inline taskLogger/redis/StepLogger.js:75]: same thing here - lets rename the field in redis too

**Review by daniel-codefresh — CHANGES_REQUESTED** [review]

File: taskLogger/firebase/StepLogger.js (hunk 4 above)
Comment (kim-codefresh) [reply taskLogger/firebase/StepLogger.js:111]: oh right

**Review by daniel-codefresh — APPROVED** [review]

**Pull request merged**